common/metrics/src/recorder.rs

```diff
@@ -87,7 +87,9 @@ fn init_prometheus_recorder() {
     let recorder = PrometheusBuilder::new().build_recorder();
     let mut h = PROMETHEUS_HANDLE.as_ref().write();
     *h = Some(recorder.handle());
-    metrics::clear_recorder();
+    unsafe {
```

Comment:
Why need unsafe here?

Reply:
https://github.com/metrics-rs/metrics/pull/302

Reply:
> `metrics::clear_recorder();` was removed from metrics (technically just marked unsafe and hidden from docs) due to being dangerous to use.

There is a potential data race here, and I think we can open an issue and try to fix it.